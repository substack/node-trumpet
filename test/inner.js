var path = require('path');
var fs = require('fs');
var trumpet = require('../');
var test = require('tap').test;
var concat = require('concat-stream');
var expected = fs
	.readFileSync(path.join(__dirname, 'inner_expected.html'))
	.toString();
var indexPage = fs
	.createReadStream(path.join(__dirname, 'inner_main.html'));
var transform = trumpet();
var elementStream = transform.select('#hello-world')
	.createWriteStream();

test('Usage of trumpet inside it\'s element write stream', function(t){
	t.plan(1);
	elementStream.on('drain', function () {
		var helloWorldPage = fs
			.createReadStream(path.join(__dirname, 'inner_block.html'));
		var innerTransform = trumpet();

		var innerElement = innerTransform
			.select('#inner-hello-world')
			.createWriteStream();

		innerElement.on('drain', function () {
			var innerHelloPage = fs
				.createReadStream(path.join(__dirname, 'inner_inner_block.html'));
			innerHelloPage.pipe(innerElement);
		});

		helloWorldPage
			.pipe(innerTransform)
			.pipe(elementStream);
	});

	indexPage
		.pipe(transform)
		.pipe(concat(function(result){
			var resultString = result.toString();
			t.deepEqual(resultString, expected);
			t.end();
		}));
});
