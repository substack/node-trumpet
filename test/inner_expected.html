<!DOCTYPE html>
<html>
<head>
    <title>Trumpet inner usage test</title>
</head>
<body>
    <div id="hello-world">
        <h1>Hello, world!</h1>
        <div id="inner-hello-world">
            <h2>Inner hello!</h2>
        </div>
    </div>
</body>
</html>